src/ITensorNetworks.jl

```diff
@@ -33,6 +34,7 @@ using TimerOutputs
 
 using DataGraphs: IsUnderlyingGraph, edge_data_type, vertex_data_type
 using Graphs: AbstractEdge, AbstractGraph, Graph, add_edge!
+using NamedGraphs: copy_keys_values
```

Comment:
By the way, it looks like the `Dictionaries.jl` copying issue was fixed: https://github.com/andyferris/Dictionaries.jl/pull/136. We'll have to bump support for `Dictionaries` to `0.4` by changing the compat entry here: https://github.com/mtfishman/ITensorNetworks.jl/blob/v0.4.0/Project.toml#L40 in order to use it.

Reply:
Could you please bump it in ITensors?

Reply:
Oh right, I forgot about that. Sure thing.

Reply:
I updated `NamedGraphs`, `DataGraphs`, `NDTensors`, and `ITensors` to use `Dictionaries` v0.4, and removed `copy_keys_values`.

Reply:
The new version of `ITensors` is still being registered but should be done soon: https://github.com/JuliaRegistries/General/pull/99896

Reply:
seems like there's still a conflict with IsApprox in ITensorNetworks (my dev'd ITensors is updated and works)
ERROR: Unsatisfiable requirements detected for package Dictionaries [85a47980]:
 Dictionaries [85a47980] log:
 ├─possible versions are: 0.1.0-0.4.0 or uninstalled
 ├─restricted to versions 0.4 by ITensors [9136182c], leaving only versions: 0.4.0
 │ └─ITensors [9136182c] log:
 │   ├─possible versions are: 0.3.55 or uninstalled
 │   └─ITensors [9136182c] is fixed to version 0.3.55
 └─restricted by compatibility requirements with IsApprox [28f27b66] to versions: 0.2.0-0.3.29 — no versions left
   └─IsApprox [28f27b66] log:
     ├─possible versions are: 0.1.0-0.1.7 or uninstalled
     ├─restricted to versions 0.1 by ITensors [9136182c], leaving only versions: 0.1.0-0.1.7
     │ └─ITensors [9136182c] log: see above
     └─restricted to versions 0.1.7-0.1 by ITensorNetworks [2919e153], leaving only versions: 0.1.7
       └─ITensorNetworks [2919e153] log:
         ├─possible versions are: 0.4.0 or uninstalled
         └─ITensorNetworks [2919e153] is fixed to version 0.4.0

Reply:
I see, I guess `IsApprox` needs to update its compat entry for `Dictionaries.jl` to support v0.4.

Reply:
Yes, but ITensors also has IsApprox v0.1 as a dependency --- why is the conflict not showing up for ITensors?

Reply:
I see, ITensors sets it to v0.1 while ITensorNetworks sets it v0.1.7, so more restrictive. Switching to v0.1 for ITensorNetworks seems to work (assuming it's not breaking something else).

Reply:
That sounds fine for now, while we wait for a fix for `IsApprox.jl` (https://github.com/jlapeyre/IsApprox.jl/pull/8).